op-challenger/game/fault/types/position.go

```diff
@@ -1,56 +1,89 @@
 package types
 
-import "fmt"
+import (
+	"fmt"
+	"math/big"
+)
 
 // Position is a golang wrapper around the dispute game Position type.
 type Position struct {
 	depth        int
-	indexAtDepth int
+	indexAtDepth *big.Int
 }
 
-func NewPosition(depth, indexAtDepth int) Position {
-	return Position{depth, indexAtDepth}
+func NewPosition(depth int, indexAtDepth big.Int) Position {
+	return Position{
+		depth:        depth,
+		indexAtDepth: &indexAtDepth,
+	}
+}
+
+func NewLargePositionFromGIndex(x *big.Int) Position {
+	depth := BigMSB(x)
+	indexAtDepth := big.NewInt(0).Sub(x, big.NewInt(0).Lsh(big.NewInt(1), uint(depth)))
+	return NewPosition(depth, *indexAtDepth)
 }
 
+// todo(refcell): remove this to use the NewLargePositionFromGIndex.
 func NewPositionFromGIndex(x uint64) Position {
 	depth := MSBIndex(x)
 	indexAtDepth := ^(1 << depth) & x
-	return NewPosition(depth, int(indexAtDepth))
+	return NewPosition(depth, *big.NewInt(int64(indexAtDepth)))
+}
+
+func (p Position) Equals(other Position) bool {
+	return p.depth == other.depth && p.indexAtDepth.Cmp(other.indexAtDepth) == 0
```

Comment:
As noted on the test, I wouldn't add this production code just to support the test. Particularly because `Equals` isn't automatically used for comparison by go so it potentially causes confusion.